src/com/zombiecastlerush/util/Parser.java

```diff
@@ -0,0 +1,23 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Parser {
+    public static void main(String[] args) throws Exception {
+        new Parser().parse("go North");
+    }
+
+    public static ArrayList<String> parse(String input){
+        ArrayList<String> actions = new ArrayList<>(Arrays.asList("go", "look", "pick-up", "drop"));
+        ArrayList<String> commands = new ArrayList<>(Arrays.asList(input.split(" ")));
+        return !actions.contains(commands.get(0)) || commands.size() == 0 ? null : reduceArray(commands);
```

Comment:
`commands.size() == 0` is not needed here. 
Also I think stylistically, it would be better to write a ternary that does the positive path if the statement evaluates to True, rather than the other way around.

So like this: `return actions.contains(commands.get(0)) ? reduceArray(commands) : null;`

Reply:
updated line of code